pkg/kv/kvpb/api.proto

```diff
@@ -2993,9 +2972,23 @@ message Header {
 
 message WriteOptions {
   uint32 origin_id = 1[(gogoproto.customname) = "OriginID"];
-  // OriginTimestamp is bound to the MVCCValueHeader of written key in the
-  // batch. Note that a kv client cannot set this if they use CPut's origin
-  // timestamp arg.
+  // OriginTimestamp is used by logical data replication. It is the MVCC
+  // timestamp of the row in the source cluster.
+  //
+  // If set the write will only succeed if the provided timestamp is greater than
+  // the OriginTimestamp field in the MVCCValueHeader of the existing value (if
+  // the value has a non-zero OriginTimestamp) or the MVCC timestamp of the
+  // existing value. If there is no existing value, the timestamp check succeeds.
```

Comment:
Should we note that this doesn't apply to IncrementRequest?  I think it is fine for our purposes not to handle Increment.